replication/binlogstreamer.go

```diff
@@ -84,11 +84,24 @@ func (s *BinlogStreamer) closeWithError(err error) {
 	}
 }
 
-func newBinlogStreamer() *BinlogStreamer {
+func NewBinlogStreamer() *BinlogStreamer {
 	s := new(BinlogStreamer)
 
 	s.ch = make(chan *BinlogEvent, 10240)
 	s.ech = make(chan error, 4)
 
 	return s
 }
+
+func (s *BinlogStreamer) AddEventToStreamer(ev *BinlogEvent) error {
+	select {
+	case s.ch <- ev:
+		return nil
+	case err := <-s.ech:
+		return err
+	}
+}
+
+func (s *BinlogStreamer) AddErrorToStreamer(err error) {
+	s.ech <- err
```

Comment:
it's a bit dangerous that s.ech is full so it will be blocked. we can use `select` to avoid blocking. And if the developer cares about the success of appending we can add a bool return value